atd-vzv/src/views/nav/CrashTypeSelector.js

```diff
@@ -0,0 +1,142 @@
+
+const CrashTypeSelector = ({ setCrashType }) => {
+  const fatalities = {
+    name: "fatalities",
+    textString: "Fatalities",
+    queryStringCrash: "(death_cnt > 0)",
+    queryStringDemographics: "(prsn_injry_sev_id = 4)"
+  };
+
+  const seriousInjuries = {
+    name: "seriousInjuries",
+    textString: "Serious Injuries",
+    queryStringCrash: "(sus_serious_injry_cnt > 0)",
+    queryStringDemographics: "(prsn_injry_sev_id = 1)"
+  };
+
+  const [activeTab, setActiveTab] = useState([fatalities, seriousInjuries]);
+  const [previousTabClicked, setPreviousTabClicked] = useState([]);
+
+  const toggle = tab => {
+    // Make a copy of the activeTab array that exists in state in order to mutate it
+    let placeHolder = [...activeTab];
+    // Attempt to filter out the object in the activeTab copy that matches the clicked tab
+    let filteredObject = placeHolder.filter(
+      objectChild => objectChild.name === tab.name
+    );
+    // Store the previous object filtered out to default back to if user clicks same tab twice
+    setPreviousTabClicked(filteredObject);
+    // If the clicked tab object is filtered out of the activeTab copy,
+    // find any remaining objects and place them into a new array
+    if (filteredObject.length) {
+      let placeHolderChild = [];
+      placeHolder.forEach(item => {
+        if (item.name !== tab.name) {
+          placeHolderChild.push(item);
+        }
+      });
+      // If there are remaining objects, set them to replace the activeTab copy, clearing out the clicked tab,
+      // else if there are no remaining objects, set the previous tab to replace the activeTab copy
+      // (toggling back to previous state rather than emptying activeTab completely and returning no results)
+      placeHolder = placeHolderChild.length
+        ? placeHolderChild
+        : previousTabClicked;
+      // If the clicked tab object was not found in the activeTab copy,
+      // add it back into the activeTab copy
+    } else {
+      placeHolder.push(tab);
+    }
+    // Set the activeTab copy to replace activeTab in state
+    setActiveTab(placeHolder);
+  };
+
+  const isUnselected = tab => {
+    let filteredObject = activeTab.find(element => element.name === tab.name);
+    let unselected = filteredObject ? false : true;
+    return unselected;
+  };
+
+  useEffect(() => {
+    const fatalitiesAndSeriousInjuries = {
+      name: "fatalitiesAndSeriousInjuries",
+      textString: "Fatalities and Serious Injuries",
+      queryStringCrash: "(death_cnt > 0 OR sus_serious_injry_cnt > 0)",
+      queryStringDemographics:
+        "(prsn_injry_sev_id = 4 OR prsn_injry_sev_id = 1)"
+    };
+
+    const handleCrashType = () => {
+      let selectedCrashType =
+        activeTab.length > 1 ? fatalitiesAndSeriousInjuries : activeTab[0];
+      return selectedCrashType;
+    };
+
+    setCrashType(handleCrashType());
+  }, [setCrashType, activeTab]);
+
+  const Button = styled.button`
```

Comment:
Just a note, I've been finding myself using `styled-components` to wrap `reactstrap` components in order to build off existing Bootstrap components or even override out-of-the-box Bootstrap classes since these components generate elements with Bootstrap classes under the hood. Overkill for this component, but I wanted to share. Handy for when `reactstrap` components aren't doing quite what you want.

Styled component
```
const StyledButton = styled.div`
    .my-style {
    /* Styles for reactstrap Button component below */
    }

    .btn {
    /* Do stuff to modify default btn styles */
    }
`;
```

JSX
```
<StyledButton>
   <Button className="my-style">primary</Button>
</StyledButton>
```

Reply:
This is super helpful! This is what I initially was trying to do before I decided to forgo Bootstrap since this was a relatively simple component. This will be really useful in the future!